### PR TITLE
[s]Wizards no longer get a free use of a touch spell by refunding the spell.

### LINKED ISSUE #46202: Wizards can get a "free" one-use version of the smite spell
Steps to reproduce:
1. Purchase the Smite spell.
2. Cast the Smite spell, but don't use it on a target yet.
3. Refund the Smite spell.
4. The next person you click on will be gibbed, as per the normal Smite spell. Note that unless you return to the wizard den, you can't recast the Smite spell after you've refunded it, so it's basically just an unrechargable version of the Smite spell.

### PR DESCRIPTION
**Pull request by harmonyn**
## About The Pull Request

If a wiz refunds a touch spell (ei nath, etc.), if there is an active touch attack, that attack is deleted. EDIT: Now with new and improved coding

Fixes https://github.com/tgstation/tgstation/issues/46202.

## Why It's Good For The Game

Fixes an exploit.

### CODE REVIEW

**Comment by bgobandit** [comment]
travis what the fuck

**Comment by ATH1909** [comment]
Thanks, man.

**Comment by ExcessiveUseOfCobblestone** [comment]
https://github.com/tgstation/tgstation/blob/2029163d335356dc5f78ddf04f0750c4fa06c429/code/modules/spells/spell_types/godhand.dm#L53

```DM
if(!attached_spell && !(flags_1 & ADMIN_SPAWNED_1)) //No Spell, No Use (Unless Admin Bafoonery)
```

**Comment by bgobandit** [comment]
ok yeah that's a way better fix

**Comment by PKPenguin321** [comment]
Actually I think the best method would be to tie removing the touch spell object into the removal proc of the spell action button

**Review by PKPenguin321 — CHANGES_REQUESTED** [review]
See above

**Comment by bgobandit** [comment]
good call

File: code/modules/spells/spell_types/touch_attacks.dm [hunk 1]
```diff
@@ -7,6 +7,12 @@
 	include_user = TRUE
 	range = -1
 
+/obj/effect/proc_holder/spell/targeted/touch/Destroy()
+	for(var/obj/item/melee/touch_attack/T in usr.contents)
+		if (istype(T, hand_path))
+			qdel(T)
+			to_chat(usr, "<span class='notice'>The power of the spell dissipates from your hand.</span>")
+
```
Comment (PKPenguin321) [inline code/modules/spells/spell_types/touch_attacks.dm]: ..()

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 2]
```diff
@@ -90,6 +90,7 @@
 			qdel(S)
 			return cost * (spell_levels+1)
 	return -1
+
```
Comment (PKPenguin321) [inline code/modules/antagonists/wizard/equipment/spellbook.dm]: run `git checkout upstream/master -- code/modules/antagonists/wizard/equipment/spellbook.dm` to totally undo all changes to this file

File: code/modules/spells/spell_types/touch_attacks.dm [hunk 3]
```diff
@@ -7,6 +7,12 @@
 	include_user = TRUE
 	range = -1
 
+/obj/effect/proc_holder/spell/targeted/touch/Destroy()
+	for(var/obj/item/melee/touch_attack/T in usr.contents)
+		if (istype(T, hand_path))
+			qdel(T)
```
Comment (PKPenguin321) [inline code/modules/spells/spell_types/touch_attacks.dm]: no need for any this, there's already a proc to handle it; run `remove_hand()`

**Pull request merged**